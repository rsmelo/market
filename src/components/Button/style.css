.button {
  font-family: var(--font-family);
  cursor: pointer;
  font-size: var(--font-size-default);
  text-align: center;
  border-radius: 4px;
  outline: 0;
  padding: 12px 14px;
  transition: 0.2s ease-out;
  border-style: solid;
  border-width: 1px;
  text-transform: uppercase;
}

.flat-green,
.flat-grey {
  color: var(--color-white);
}

.flat-green {
  background: var(--color-light-green);
  border-color: var(--color-light-green);
}

.flat-green:hover {
  background: var(--color-dark-green);
  border-color: var(--color-dark-green);
}

.flat-grey {
  background: var(--color-ligther-grey);
  border-color: var(--color-ligther-green);
}

.flat-grey:hover {
  background-color: var(--color-ligth-grey);
  border-color: var(--color-light-ligthgreen);
}

.outline {
  background: transparent;
}

.outline-green {
  color: var(--color-light-green);
  border-color: var(--color-light-green);
}

.outline-green:hover {
  color: var(--color-dark-green);
  border-color: var(--color-dark-green);
}

.outline-grey {
  color: var(--color-ligther-grey);
  border-color: var(--color-ligther-grey);
}

.outline-grey:hover {
  color: var(--color-ligth-grey);
  border-color: var(--color-ligth-grey);
}

.button:disabled:hover,
.button:disabled {
  background: var(--color-ligther-grey);
  color: var(--color-ligth-grey);
  border-color: var(--color-ligther-grey);
  cursor: not-allowed;
}
