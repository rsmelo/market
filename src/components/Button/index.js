import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import style from './style.css'

const Button = ({
  onClick,
  type,
  children,
  color,
  appearance,
}) => {
  const styles = classNames(
    style.button,
    style[appearance],
    style[`${appearance}-${color}`]
  )

  return (
    <button
      className={styles}
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  )
}


Button.defaultProps = {
  onClick: null,
  appearance: 'flat',
  color: 'green',
  type: 'button',
}

Button.propTypes = {
  type: PropTypes.oneOf(['submit', 'reset', 'button']),
  onClick: PropTypes.func,
  appearance: PropTypes.oneOf([
    'flat', 'outline',
  ]),
  color: PropTypes.oneOf([
    'green', 'grey',
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
}

export default Button
