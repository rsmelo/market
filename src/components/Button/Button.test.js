import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Button from './index'

describe('Button', () => {
  it('should render correctly', () => {
    const onClick = jest.fn()
    const tree = renderer
      .create(<Button onClick={onClick} >Test</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should be clicked', () => {
    const onClick = jest.fn()
    const component = shallow(<Button onClick={onClick} >Test</Button>)
    component.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
