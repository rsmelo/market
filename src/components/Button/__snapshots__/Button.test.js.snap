// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Test
</button>
`;
